### PR TITLE
Feature/fct2 21384 selectable bucket

### PR DESCRIPTION
**Pull request by bridgesr**
# PR checklist

Tick what applies before you raise. Delete anything that doesn't.

## Correctness
- [x] Does what the ticket asks for
- [x] Handles the edge cases (empty, null, zero, large inputs), not just the happy path
- [x] Errors are handled, not swallowed
- [x] New logic has tests, including the failure cases
- [x] Tests assert something real, not just run the code

## Keeping it simple
- [x] Doesn't rebuild something that already exists in the codebase
- [x] No more complex or slower than it needs to be
- [x] No leftover debug logging or commented-out code

## Code Quality
- [x] Pre-commit hooks were run for all commits in this PR

## Easy to miss (a green pipeline won't flag these)
- [x] One logical change, not a pile of unrelated stuff

### If you touched the backend
- [x] Schema changes have a migration, and the Down() doesn't drop anything it shouldn't
- [x] New app settings are wired into the fa-config templates
- [x] Secrets use Key Vault references, not plain text
- [x] New outbound HTTP clients go through the shared resilience handler (AddResiliencePolicyHandler), rather than hand-rolling retries
- [x] New calls to an external service (DDEI, Egress, NetApp) have client tests against a WireMock stub (for the serialisation, auth and error handling) and are covered by the integration tests that run against the real dev services

### If you touched the UI
- [x] Ran the e2e/ suite locally (the PR build only runs the mocked tests; the full suite runs post-merge and gates the deploy)
- [x] API changes are reflected in the MSW handlers in src/mocks and the matching zod schema, so the mocked tests aren't passing against a contract that's gone
- [x] New config or feature flags are wired into the VITE build variables, not just added in code
- [x] Data fetches handle the loading, empty and error states, not just success, and match how the rest of the app already does it
- [x] Ran prettier (npm run prettier:format), since CI lints but doesn't check formatting
- [x] Checked accessibility (keyboard nav, screen reader, sensible labels), since the Lighthouse run is manual and won't flag it for you

### CODE REVIEW

File: backend/CPS.ComplexCases.API/Functions/SearchNetAppFolders.cs
Comment (HasanCPS) [inline backend/CPS.ComplexCases.API/Functions/SearchNetAppFolders.cs]: The `MaterialBatchRename.cs` file also needs to be updated to follow this pattern. The corresponding test files (`MaterialRenameTests.cs` and `MaterialBatchRenameHelperTests.cs`) should also be updated.

File: backend/CPS.ComplexCases.API/Functions/ListNetAppFiles.cs [hunk 1]
```diff
@@ -45,9 +48,17 @@ public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous,
         var take = int.TryParse(req.Query[InputParameters.Take], out var takeValue) ? takeValue : 100;
         var path = req.Query[InputParameters.Path];
 
-        var securityGroups = await _securityGroupMetadataService.GetUserSecurityGroupsAsync(context.BearerToken);
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.API/Functions/ListNetAppFiles.cs]: The ticket states that "list and browse endpoints need a bucket before the case has a persisted value (i.e. during the connect wizard)".

`ListNetAppFolders` correctly supports both the `caseId` and `bucketName` query parameters, whereas `ListNetAppFiles` only supports `caseId`.

As a result, if a user needs to preview files during the connect wizard before a case has a persisted bucket, `ResolveBucketAsync` is called with `(bearerToken, null, null)` and silently falls back to `.First()`.

This reintroduces the ordering bug for this code path, as bucket selection becomes dependent on the order of the returned results rather than an explicitly specified bucket.

Also add the corresponding `[OpenApiParameter]` attribute and update `ListNetAppFilesTests`.

**Review by HasanCPS — APPROVED** [review]
LGTM!
